core/store/src/trie/trie_storage.rs

```diff
@@ -208,16 +207,15 @@ impl TrieCacheInner {
 pub struct TrieCache(Arc<Mutex<TrieCacheInner>>);
 
 impl TrieCache {
-    pub fn new(shard_id: u32, is_view: bool) -> Self {
-        Self::with_capacities(TRIE_DEFAULT_SHARD_CACHE_SIZE, shard_id, is_view)
-    }
-
-    pub fn with_capacities(cap: usize, shard_id: u32, is_view: bool) -> Self {
+    pub fn new(config: &TrieConfig, shard_uid: ShardUId, is_view: bool) -> Self {
+        let capacity = config.shard_cache_capacity(shard_uid, is_view);
+        let total_size_limit = config.shard_cache_total_size_limit(shard_uid, is_view);
+        let queue_capacity = config.deletions_queue_capacity();
         Self(Arc::new(Mutex::new(TrieCacheInner::new(
-            cap,
-            DEFAULT_SHARD_CACHE_DELETIONS_QUEUE_CAPACITY,
-            DEFAULT_SHARD_CACHE_TOTAL_SIZE_LIMIT,
-            shard_id,
+            capacity as usize,
+            queue_capacity,
+            total_size_limit,
+            shard_uid.shard_id(),
```

Comment:
Consider adding a `TrieCacheConfig` with `capacity, total_size_limit, queue_capacity`, passing in just that, and removing `shard_id` and `is_view`parametres altogether (we only need them for `log_assert`, and I am fairly sure we can live without this info there).


Ahhh, no we need those to create metrics as well. :thinking:, yeah, I think this is ok then.